src/Symfony/Component/Security/Core/Authentication/Provider/LdapBindAuthenticationProvider.php

```diff
@@ -84,7 +84,7 @@ protected function checkAuthentication(UserInterface $user, UsernamePasswordToke
         $username = $token->getUsername();
         $password = $token->getCredentials();
 
-        if ('' === $password) {
+        if (empty($password)) {
```

Comment:
We try to avoid using `empty` in Symfony.  Here, a password like `0` would return `true` with empty, which is not what we want here.

Reply:
Thanks for feedback, What about use the `NotBlankValidator` condition `if(empty($value) && '0' != $value)`

Reply:
I'd suggest to write this as `if ('' === (string) $password) {`

Reply:
Thanks @nicolas-grekas I applied your code in the PR, Hope it's good